nni/runtime/config.py

```diff
@@ -1,13 +1,18 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
+from datetime import datetime
 import os
 from pathlib import Path
 import shutil
 import sys
 
 import nni
 
+# the version when default config got modified
+# don't need to update if default config does not change
+latest_config_version = '2.6'
```

Comment:
Suggest reading `registered_algorithms.yml` and putting its md5sum here.

Reply:
`registered_algorithms.yml` can be modified by user with `nnictl algo register`.
This is only relevant to `default_config/registered_algorithms.yml` committed into our code repo.

Reply:
I meant, why can't we automatically calculate and check the checksum here instead of a string `2.6`?

Reply:
Well...... The config can be updated with `nnictl algo register`, so we do not except `~/.config/nni/registered_algorithms.yml` to be identical to `nni/runtime/default_configs/registered_algorithms.yml`.

Reply:
We store a checksum in `nni/runtime.config.py`. When users access their own `registered_algorithms.yml`, we check the checksum which was stored as a field in the yaml,

1. If a match -- no problem
2. Otherwise -- we update his/her yaml, as well as the checksum in the yaml.

As for updating from more than one old formats, seemingly we might as well use `nni.__version__` as checksum. But my concern is that `nni.__version__` might be unreliable in develop mode. Why don't we keep another field called schema version?

I'm against writing the version **manually** here, because people **forget**.

Reply:
I agree, manually writing version number here is not a good practice